mfr/extensions/jsc3d/export.py

```diff
@@ -0,0 +1,42 @@
+import os
+import subprocess
+
+import shutil
+from tempfile import NamedTemporaryFile
+from mfr.core import extension
+from mfr.core import exceptions
+from mfr.extensions.jsc3d import settings
+
+
+class JSC3DExporter(extension.BaseExporter):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def export(self):
+        try:
+            # alter original source file with a named temp file, so freecad can read it properly
```

Comment:
What's the reason for `freecad` not being able to read it? (e.g. the extension name, the permissions, the path, etc.) When is the temporary file removed?

Reply:
Since the temp file is opened with at `with` clause, once it leaves the `with` it will close. The output file is written at this point, so it doesn't need to be open anymore

The reason for the temp file is that mfr gives the file without the extension, and freecad opens files based on the extension. I have to add it back in for it to be able to open it.